### PR TITLE
MoveitStateAdaptor getIK joint limit check

### PR DESCRIPTION
**Pull request by jrgnicho**
This PR modifies the **getIK** method in order to verify that the joint solution is within the joint limits as defined in the **URDF** file for the robot model.

### CODE REVIEW

**Comment by jrgnicho** [comment]
@shaun-edwards please review

File: descartes_moveit/src/moveit_state_adapter.cpp [hunk 1]
```diff
@@ -145,9 +145,9 @@ bool MoveitStateAdapter::getIK(const Eigen::Affine3d &pose, std::vector<double>
                               tool_frame_))
   {
     robot_state_->copyJointGroupPositions(group_name_, joint_pose);
-    if(isInCollision(joint_pose))
+    if(!isValid(joint_pose))
```
Comment (shaun-edwards) [inline descartes_moveit/src/moveit_state_adapter.cpp:148]: Does `isValid` check for collisions?

File: descartes_moveit/src/moveit_state_adapter.cpp [hunk 2]
```diff
@@ -109,7 +109,7 @@ bool MoveitStateAdapter::initialize(const std::string& robot_description, const
     if (world_frame_ != robot_state_->getRobotModel()->getModelFrame())
     {
       logWarn("World frame '%s' does not match model root frame '%s', all poses will be transformed to world frame '%s'",
-               world_frame_.c_str(), link_names.front().c_str(),world_frame_.c_str());
+               world_frame_.c_str(), robot_state_->getRobotModel()->getModelFrame().c_str(),world_frame_.c_str());
```
Comment (shaun-edwards) [inline descartes_moveit/src/moveit_state_adapter.cpp:112]: Just a suggestion, but if we change log messages, we should also take the opportunity to update to the ROS macros.

File: descartes_moveit/src/moveit_state_adapter.cpp (hunk 1 above)
Comment (shaun-edwards) [inline descartes_moveit/src/moveit_state_adapter.cpp:148]: Yep...it appears it does ;)

**Comment by shaun-edwards** [comment]
+1...you can merge.

**Comment by jrgnicho** [comment]
It does check for collisions.  I'll update the macros.  Thanks

**Comment by jrgnicho** [comment]
I replaced the log messages for the equivalent ROS macros but started getting lots of warnings during the build.  It looks like this might take more than a simple replacement so I created an issue for it here #116.  I'll address that in a different PR.

**Pull request merged**